Makefile

```diff
@@ -0,0 +1,95 @@
+	conan profile update settings.build_type=$(build_type) $(conan_profile_name)
+	conan remove -f "pybind11/*"
+	conan create --profile=$(conan_profile_name) recipes/pybind11
+
+ # Currently the b2 package (a dependency of boost) in the remote tket-libs does not include compiler info
+ # which can lead to errors when building boost from source. Conan will try to use
+ # the remote binary even if it is incompatible with the compiler, leading to linking issues.
+ # This fixes that problem by forcing a b2 build from source
+.PHONY: b2
+b2: # install and build b2 (for boost builds) from source
+	conan profile update settings.build_type=$(build_type) $(conan_profile_name)
+	conan install --profile=$(conan_profile_name) b2/4.9.2@ --build
+
+.PHONY: local-libs
+local-libs: # create conan packages for local libs from this repository
+	conan profile update settings.build_type=$(build_type) $(conan_profile_name)
+	conan create --profile=$(conan_profile_name) libs/tklog tket/stable --build=missing
+	conan create --profile=$(conan_profile_name) libs/tkassert tket/stable --build=missing
+	conan create --profile=$(conan_profile_name) libs/tkrng tket/stable --build=missing
+	conan create --profile=$(conan_profile_name) libs/tktokenswap tket/stable --build=missing
+	conan create --profile=$(conan_profile_name) libs/tkwsm tket/stable --build=missing
+
+.PHONY: dev-env-from-scratch
+dev-env-from-scratch: b2 local-libs ## Install tket, tket-tests, and tket-proptests into a local build directory and configure cmake (build everything from source)
+	conan profile update settings.build_type=$(build_type) $(conan_profile_name)
+	conan install --profile=$(conan_profile_name) recipes/tket  --install-folder=$(conan_build_dir)/tket/src --build=missing
+	conan install --profile=$(conan_profile_name) recipes/tket-tests --install-folder=$(conan_build_dir)/tket/tests --build=missing
+	conan install --profile=$(conan_profile_name) recipes/tket-proptests --install-folder=$(conan_build_dir)/tket/proptests --build=missing
+	cmake -DCMAKE_BUILD_TYPE=$(build_type) -DCMAKE_MAKE_PROGRAM=ninja -G Ninja -S . -B $(conan_build_dir)
+
+.PHONY: dev-env
+dev-env: ## Install tket, tket-tests, and tket-proptests into a local build directory and configure cmake (dependencies from cache or remote)
+	conan profile update settings.build_type=$(build_type) $(conan_profile_name)
+	conan install --profile=$(conan_profile_name) recipes/tket  --install-folder=$(conan_build_dir)/tket/src
+	conan install --profile=$(conan_profile_name) recipes/tket-tests --install-folder=$(conan_build_dir)/tket/tests
+	conan install --profile=$(conan_profile_name) recipes/tket-proptests --install-folder=$(conan_build_dir)/tket/proptests
+	cmake -DCMAKE_BUILD_TYPE=$(build_type) -DCMAKE_MAKE_PROGRAM=ninja -G Ninja -S . -B $(conan_build_dir)
+
+##@ Build
+
+.PHONY: build
+build: ## build with cmake
+	cmake --build $(conan_build_dir) --target test_tket -j 6
+	cmake --build $(conan_build_dir) --target proptest -j 6
```

Comment:
Instead of "6" we could make this configable, defaulting to `nproc`?

Reply:
I configured `n_cpus` using `python -c 'import multiprocessing as m; print(m.cpu_count())'`, which I think should work as long as python is installed, also on mac and windows